pyanaconda/modules/payloads/payload/live_image/installation.py

```diff
@@ -445,17 +440,42 @@ def _install_image(self):
         ]
 
         try:
-            rc = execWithRedirect(cmd, args)
+            self.report_progress(_("Installing software..."))
+            reader = execReadlines(cmd, args)
+            for line in reader:
+                self._parse_rsync_update(line)
+
         except (OSError, RuntimeError) as e:
             msg = "Failed to install image: {}".format(e)
             raise PayloadInstallationError(msg) from None
 
-        if rc == 11:
+        if reader.rc == 11:
             raise PayloadInstallationError(
                 "Failed to install image: "
-                "{} exited with code {}".format(cmd, rc)
+                "{} exited with code {}".format(cmd, reader.rc)
             )
 
+    def _parse_rsync_update(self, line):
+        """Try to extract progress from rsync output.
+
+        :param str line: line to process
+        """
+        # Take only part after last ^M (python: \r)
+        # rsync overwrites progress line, and the interesting data is there only last time.
```

Comment:
I think there is some sort of a typo or at least I find it hard to read. :)

Reply:
Ah, true. Probably needs more text to describe what's happening.